### ISSUE TITLE
SyncOnSubscribeTest.testUnsubscribeOutsideOfLoop FAILED

### ISSUE THREAD

**Issue by akarnokd** [ReactiveX/RxJava#3281]
@stealthcode 
```
rx.observables.SyncOnSubscribeTest > testUnsubscribeOutsideOfLoop FAILED
    java.lang.AssertionError: Not unsubscribed.
        at rx.observers.TestSubscriber.assertUnsubscribed(TestSubscriber.java:250)
        at rx.observables.SyncOnSubscribeTest.testUnsubscribeOutsideOfLoop(SyncOnSubscribeTest.java:645)
```

**Comment by stealthcode** [comment]
When does this fail? That test passes on my machine.

**Comment by stealthcode** [comment]
I got it to fail after 2969 iterations. I'll fix it.

**Comment by stealthcode** [comment]
See #3285. Thanks

**Comment by akarnokd** [comment]
Fixed via #3285
